### PR TITLE
Disconnect: if Rewind is active, prompt user to try rewind before disconnecting

### LINKED ISSUE #22364: Site disconnect: modify disconnection screen for sites with backups
As pointed out here hp9rlnk-7E if a user with backups disconnects their site from Calypso, they will lose access to their backups. In order to prevent that, I propose modifying the disconnect dialogue to include a button directing folks to the activity log and a happy chat button to get help.

image

image

image

The only image above modified is the last. It has an added paragraph at the top and the two buttons.

* The buttons should both be tracked.
* If Happiness gets overwhelmed, we should hide the button.

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #22364

This PR introduces a new box prompting user to try Rewind to fix their site when they're having issues with it:

- introduces a button to go to Activity Log and another to launch a Happychat
- both buttons have tracking
- implements the copy fixes suggested by @clucasrowlands (this plan is different from the one used for the issue but the code shows that all issues have been addressed)
- declares the property `primary` in the `HappychatButton` component. It was already in use, it just wasn't declared

captura de pantalla 2018-02-22 a la s 18 14 52

#### Testing

In a connected Jetpack site that has Rewind active, go to the disconnection dialog and verify things look good.

### CODE REVIEW

File: client/blocks/disconnect-jetpack/index.jsx [hunk 1]
```diff
@@ -257,22 +260,28 @@ class DisconnectJetpack extends PureComponent {
 					className="disconnect-jetpack__more-info-link"
 					href="https://jetpack.com/features/"
 				>
-					{ translate( 'Read More about Jetpack benefits' ) }
+					{ translate( 'Read more about Jetpack benefits' ) }
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: :ok: This change is safe, we'll keep the existing 21 translations.

File: client/blocks/disconnect-jetpack/index.jsx [hunk 2]
```diff
@@ -257,22 +264,42 @@ class DisconnectJetpack extends PureComponent {
...
+				<Card
+					key="disconnect-jetpack__try-rewind"
+					className="disconnect-jetpack__try-rewind disconnect-jetpack"
+				>
+					<p className="disconnect-jetpack__highlight">
+						{ translate( 'Experiencing connection issues? Try to go back and rewind your site.' ) }
+					</p>
+					<div className="disconnect-jetpack__try-rewind-button-wrap">
+						<Button href={ `/stats/activity/${ siteSlug }` } onClick={ this.props.trackTryRewind }>
+							{ translate( 'Rewind site' ) }
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Rewind Site' )` *ES Score: 17*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/disconnect-jetpack/index.jsx [hunk 3]
```diff
@@ -104,31 +108,31 @@ class DisconnectJetpack extends PureComponent {
...
 					translate(
-						'{{icon/}} Daily, automated backups (unlimited storage)',
+						'{{icon/}} Daily automated backups (unlimited storage)',
 						this.getIcon( 'history' )
 					)
 				);
 				features.push(
 					translate(
-						'{{icon/}} Daily, automated malware scanning with automated resolution',
+						'{{icon/}} Daily automated malware scanning with automated resolution',
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: :ok: This change is safe, we'll keep the existing 19 translations.

File: client/blocks/disconnect-jetpack/index.jsx [hunk 4]
```diff
@@ -91,7 +95,7 @@ class DisconnectJetpack extends PureComponent {
 			case 'is-personal-plan':
 				features.push(
 					translate(
-						'{{icon/}} Daily, automated backups (unlimited storage)',
+						'{{icon/}} Daily automated backups (unlimited storage)',
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: :disappointed: 19 existing translations will be lost with this change.

File: client/blocks/disconnect-jetpack/index.jsx [hunk 5]
```diff
@@ -104,31 +108,31 @@ class DisconnectJetpack extends PureComponent {
...
 				features.push(
-					translate( '{{icon/}} Daily, automated malware scanning', this.getIcon( 'spam' ) )
+					translate( '{{icon/}} Daily automated malware scanning', this.getIcon( 'spam' ) )
 				);
 				features.push(
 					translate( '{{icon/}} Priority WordPress and security support', this.getIcon( 'chat' ) )
 				);
 				features.push(
-					translate( '{{icon/}} 13Gb of high-speed video hosting', this.getIcon( 'video' ) )
+					translate( '{{icon/}} 13GB of high-speed video hosting', this.getIcon( 'video' ) )
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: :ok: This change is safe, we'll keep the existing 19 translations.

File: client/blocks/disconnect-jetpack/index.jsx [hunk 6]
```diff
@@ -104,31 +108,31 @@ class DisconnectJetpack extends PureComponent {
...
 				features.push(
 					translate(
-						'{{icon/}} Daily, automated backups (unlimited storage)',
+						'{{icon/}} Daily automated backups (unlimited storage)',
 						this.getIcon( 'history' )
 					)
 				);
 				features.push(
-					translate( '{{icon/}} Daily, automated malware scanning', this.getIcon( 'spam' ) )
+					translate( '{{icon/}} Daily automated malware scanning', this.getIcon( 'spam' ) )
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: :ok: This change is safe, we'll keep the existing 19 translations.

File: client/blocks/disconnect-jetpack/index.jsx [hunk 7]
```diff
@@ -257,32 +267,58 @@ class DisconnectJetpack extends PureComponent {
...
+				<Card
+					key="disconnect-jetpack__try-rewind"
+					className="disconnect-jetpack__try-rewind disconnect-jetpack"
+				>
+					<p className="disconnect-jetpack__highlight">
+						{ translate( 'Experiencing connection issues? Try to go back and rewind your site.' ) }
+					</p>
+					<div className="disconnect-jetpack__try-rewind-button-wrap">
+						<Button href={ `/stats/activity/${ siteSlug }` } onClick={ this.handleTryRewind }>
+							{ translate( 'Rewind site' ) }
```
Comment (a8ci18n) [inline client/blocks/disconnect-jetpack/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Rewind Site' )` *ES Score: 17*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by eliorivero** [comment]
Thanks for the reviews!
The missing chat panel seems to be happening in staging and development. Given that this uses the HappychatButton component, it's safe to use it and as long as Happychat is working fine, this will launch it.
I'll merge for now and will monitor once it's deployed.

**Comment by eliorivero** [comment]
Just tested this in production and it's working fine, opening the HappyChat panel, so all good 👍

**Pull request merged**